src/sentry/api/serializers/models/sentry_app_component.py

```diff
@@ -22,7 +25,7 @@ def serialize(self, obj, attrs, user, errors):
 
 class SentryAppAlertRuleActionSerializer(Serializer):
     def serialize(self, obj, attrs, user, **kwargs):
-        event_action = kwargs.get("event_action")
+        event_action: SentryAppEventDataInterface | None = kwargs.get("event_action")
```

Comment:
Unfortunately, `SentryAppEvent` objects mix a lot of data together that isn'tnecesary and is definitely harder to serialize.  I extracted this reduce interface that matches only what the serializer needs.